app.py

```diff
@@ -207,6 +209,28 @@ def save_identity():
     return jsonify({})
 
 
+# End Point example which verifies payload signature
+@app.route('/v1/signed_payload', methods=['POST'])
+@validate_json
+def signed_payload():
+    payload = request.get_json(force=True)
+    signature_base64_encoded = request.headers.get('signature')
+    identity = payload['identity']
+
+    signature_bytes = base64.b64decode(signature_base64_encoded)
+
+    recovered_public_address = recover_public_address(
+        request.data,
+        signature_bytes,
+    )
+
+    # verify payload
+    if recovered_public_address.lower() == identity.lower():
+        return jsonify({})
```

Comment:
Return HTTP status with errors payload. See https://httpstatuses.com/

- Identity empty
- Identity given, but malformed format
- Identity unknown in database
- Signature empty
```
401
{
  "error":  "Identity cant be empty" 
}

```
403
{
  "error": "Invalid signature"
}
```

Reply:
There is a list of predefined 4xx http errors for various cases:
400 - Bad request (most generic one), usually used to indicate that your payload is malformed
401 - Authentication required - signature invalid/missing
403 - Forbidden identity provided by this signature is valid but not registered in our databases
422 - Unprocessable entity - used to indicate that request body is syntactically correct (i.e. valid JSON/XML structure) but fields contain incomplete missing information, usually reports validation errros.

Reply:
I updated my comment here

Reply:
added 
401 - payload was not signed with provided identity